tree/treeplayer/inc/ROOT/TTreeProcessorMT.hxx

```diff
@@ -89,12 +98,17 @@ private:
    std::vector<std::string> FindTreeNames();
    static unsigned int fgTasksPerWorkerHint;
 
+   EntryRange fGlobalRange{};
+
 public:
-   TTreeProcessorMT(std::string_view filename, std::string_view treename = "", UInt_t nThreads = 0u);
+   TTreeProcessorMT(std::string_view filename, std::string_view treename = "", UInt_t nThreads = 0u,
+                    const EntryRange &globalRange = {0, std::numeric_limits<Long64_t>::max()});
    TTreeProcessorMT(const std::vector<std::string_view> &filenames, std::string_view treename = "",
-                    UInt_t nThreads = 0u);
-   TTreeProcessorMT(TTree &tree, const TEntryList &entries, UInt_t nThreads = 0u);
-   TTreeProcessorMT(TTree &tree, UInt_t nThreads = 0u);
+                    UInt_t nThreads = 0u, const EntryRange &globalRange = {0, std::numeric_limits<Long64_t>::max()});
+   TTreeProcessorMT(TTree &tree, const TEntryList &entries, UInt_t nThreads = 0u,
+                    const EntryRange &globalRange = {0, std::numeric_limits<Long64_t>::max()});
```

Comment:
we don't have to support the case of TEntryList + entry range. does this come for free or can we remove this case and simplify our lives?

```suggestion
   TTreeProcessorMT(TTree &tree, const TEntryList &entries, UInt_t nThreads = 0u,
                    const EntryRange &globalRange = {0, std::numeric_limits<Long64_t>::max()});
```

Reply:
Yes, it is totally possible. The only small difference is that I would need to introduce if conditions inside RunTreeProcessorMT() (in the RLoopManager.cxx), e.g. here:
```cpp
   const auto &entryList = fTree->GetEntryList() ? *fTree->GetEntryList() : TEntryList();
   auto tp = std::make_unique<ROOT::TTreeProcessorMT>(*fTree, entryList, fNSlots, EntryRange{fStartEntry, fEndEntry});
```

Reply:
If the user doesn't explicitly pass a TEntryList, I argue we should always have an overload without TEntryList and use always that (this overload allows the globalrange)

Reply:
agreed!